tests/e2e/awesome_functions/awesome_functions.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2023 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+import sys
+
+import pytest
+from common import memgraph
+
+
+def test_indexed_join_with_indices(memgraph):
+    def get_bytes(prop_name):
+        res = list(memgraph.execute_and_fetch(f"MATCH (n) RETURN property_size(n, '{prop_name}') AS size"))
+        return res[0]["size"]
+
+    memgraph.execute(
+        """
+        CREATE (n:Node)
+        SET n.null_prop = null,
```

Comment:
Great description of what is going on. I think it would be better to have 1 test for each property type because it would be easier to change things in the future but ok

Reply:
Yes I will separate each test into its own test so we can easily test which test failed.